src/object/obj_internal.h

```diff
@@ -156,7 +161,7 @@ static inline bool
 obj_retry_error(int err)
 {
 	return err == -DER_TIMEDOUT || err == -DER_STALE ||
-	       daos_crt_network_error(err);
+	       err == -DER_INPROGRESS || daos_crt_network_error(err);
```

Comment:
DER_INPROGRESS means just retry, not need refresh the pool map. Refresh the pool map under such case will cause overhead, but no correctness issue. On the other hand, if we directly retry, relate (conflict) operation may still not complete yet, then such retry will also be useless. So some degree, the refresh of pool map can reduce retry times.

Reply:
Hmm, bothering the server unnecessary may not be a smart way as well.  Probably the client only needs to do a few crt_progress() or "sleep" there instead of keep sending query to the leader for no reason.

Reply:
The retry is not shard level, may be client level to other shard, so only crt_progress() may be not enough.

Reply:
because you will set retry_with_leader, so I think we can skip the pool_query if it's set?